src/main/kotlin/org/rust/cargo/runconfig/buildtool/CargoBuildAdapter.kt

```diff
@@ -110,6 +110,12 @@ class CargoBuildAdapter(
         // otherwise the line contains only part of the message.
         if (concat.contains("{\"reason\"") && !concat.endsWith("}\n")) return
 
+        // Bit of a weird hack, but it's entirely possible that the error message contains explanations.
+        // Those explanations can contain "}\n", and even more bizarre, it's entirely possible that
+        // we get the text event that splits the message exactly on that boundary.
+        val count = concat.count { it == '\"' }
+        if ((count and 1) != 0) return
```

Comment:
It doesn't work for the following code:
```rust
fn main() {
    // "
    0
}
```

Reply:
Yeah, I knew it was fragile, but short of changing cargo, by requiring a NUL byte or something, we're out of luck.